docs/rules/no-callback-in-promise.md

```diff
@@ -1,3 +1,55 @@
-# Avoid calling `cb()` inside of a `then()` (use [nodeify][] instead) (no-callback-in-promise)
+# Avoid calling `cb()` inside of a `then()` or `catch()` (no-callback-in-promise)
+
+As a general rule, callbacks should never be directly invoked inside a [Promise.prototype.then()] or [Promise.prototype.catch()] method. That's because your callback may be unintentionally be invoked twice. Take the following example:
```

Comment:
So this is a great explanation of a bug that can occur when mixing these two types of patterns. My purpose for introducing this rule was more to encourage teams to pick a common pattern such as callbacks, async/await or promise and stick with it for consistencies sake, but this is a decent explanation and honestly pretty well thought out.